reactioncommerce/reaction
# feat: Update default navigation item visibility options when added to the navigation tree

Resolves #5189   
Impact: **major**  
Type: **feature**

## Issue

Navigation items a visible by default when placed in a navigation tree. This could lead to admins publishing content not yet ready for customers.

## Solution

- Default the option "Show in storefront" to `false`
- Add shop settings to override the default behavior (API only)

## Breaking changes

- More like a breaking workflow. Every nav item added to the nav tree will need to be explicitly made visible on the storefront.
- Renamed plugin in register.js from "navigation" to "reaction-navigation". Added a migration.
- Removed "registry" and "layout" config from plugin registration

## Testing
1. Add a new navigation item
2. Drag it into the tree
3. Click "Save changes"
4. See the navigation item update and say "Hide from storefront"
5. Edit the item
6. check "Show in storefront"
7. Save changes in the modal
8. Save changes into the toolbar at the top
9. Refresh the page and make sure the item doesn't say "Hide from storefront"

(all other UI hiccups will be addressed in this ticket https://github.com/reactioncommerce/reaction/issues/5163)
